src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -605,23 +606,35 @@ protected void clipForOverlappingReads(final SAMRecord read1, final SAMRecord re
                     final int negDiff = pos.getAlignmentStart() - neg.getAlignmentStart();
 
                     if (posDiff > 0) {
-                        CigarUtil.softClip3PrimeEndOfRead(pos, Math.min(pos.getReadLength(),
-                                pos.getReadLength() - posDiff + 1));
+                        final List<CigarElement> elems = new ArrayList<>(pos.getCigar().getCigarElements());
+                        Collections.reverse(elems);
+                        final int clipped = lengthOfSoftClipping(elems.iterator());
+                        final int clipFrom = pos.getReadLength() - posDiff - clipped + 1;
+                        CigarUtil.softClip3PrimeEndOfRead(pos, Math.min(pos.getReadLength(), clipFrom));
                         removeNmMdAndUqTags(pos); // these tags are now invalid!
                     }
 
                     if (negDiff > 0) {
-                        CigarUtil.softClip3PrimeEndOfRead(neg, Math.min(neg.getReadLength(),
-                                neg.getReadLength() - negDiff + 1));
+                        final int clipped = lengthOfSoftClipping(neg.getCigar().getCigarElements().iterator());
+                        final int clipFrom = neg.getReadLength() - negDiff - clipped + 1;
+                        CigarUtil.softClip3PrimeEndOfRead(neg, Math.min(neg.getReadLength(), clipFrom));
                         removeNmMdAndUqTags(neg); // these tags are now invalid!
                     }
-
                 }
-            } else {
-                // TODO: What about RR/FF pairs?
             }
         }
+    }
+
+    /** Returns the number of soft-clipped bases until a non-soft-clipping element is encountered. */
+    private static int lengthOfSoftClipping(Iterator<CigarElement> iterator) {
+        int clipped = 0;
+        while (iterator.hasNext()) {
+            final CigarElement elem = iterator.next();
+            if (elem.getOperator() != CigarOperator.SOFT_CLIP) break;
```

Comment:
Ah, good catch.  I've fixed that.  The only remaining issue is that it will lose the hard-clipping due to the issue here:
    https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/CigarUtil.java#L46

The cigar will be correctly soft-clipped, but the hard-clip will get lost.  I'm not inclined to try and fix that at the moment unless you think it's critical.